web-common/src/main/java/edu/unc/lib/boxc/web/common/services/WorkFilesizeService.java

```diff
@@ -0,0 +1,96 @@
+package edu.unc.lib.boxc.web.common.services;
+
+import edu.unc.lib.boxc.auth.api.Permission;
+import edu.unc.lib.boxc.auth.api.models.AccessGroupSet;
+import edu.unc.lib.boxc.auth.api.services.GlobalPermissionEvaluator;
+import edu.unc.lib.boxc.search.api.facets.CutoffFacet;
+import edu.unc.lib.boxc.search.api.models.ContentObjectRecord;
+import edu.unc.lib.boxc.search.api.requests.SearchRequest;
+import edu.unc.lib.boxc.search.api.requests.SearchState;
+import edu.unc.lib.boxc.search.solr.services.SolrSearchService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.math.RoundingMode;
+import java.text.DecimalFormat;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Service to retrieve the total filesize of all files in work and return a formatted String
+ *
+ * @author lfarrell
+ */
+public class WorkFilesizeService {
+    private static final Logger log = LoggerFactory.getLogger(AccessCopiesService.class);
+
+    private GlobalPermissionEvaluator globalPermissionEvaluator;
+    private SolrSearchService solrSearchService;
+
+
+    public String getTotalFilesize(ContentObjectRecord contentObjectRecord, AccessGroupSet principals) {
+        var ONE_GIGABYTE = 1073741824L;
+        var request = buildChildrenQuery(contentObjectRecord, principals);
+        var resp = solrSearchService.getSearchResults(request);
+
+        if (resp.getResultCount() > 0) {
+            var totalFileSize = resp.getResultList().stream()
+                    .mapToLong(ContentObjectRecord::getFilesizeSort)
+                    .reduce(0, Long::sum);
+
+            if (totalFileSize > ONE_GIGABYTE) {
```

Comment:
I would also recommend having this service always return the actual size, and then having either the full record controller or the vue code decide not to make the download available based on size. That way we could use the service as a way to get the full work size for other purposes, or show what the full bulk download size would be if they requested it from wilson. 

I'm also wondering how we want to handle blocking download of really big bulk downloads if the user put in the url directly, but that seems like a separate ticket.